oap-server/analyzer/log-analyzer/src/main/java/org/apache/skywalking/oap/log/analyzer/provider/log/listener/TrafficAnalysisListener.java

```diff
@@ -0,0 +1,113 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.skywalking.oap.log.analyzer.provider.log.listener;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.skywalking.apm.network.logging.v3.LogData;
+import org.apache.skywalking.apm.util.StringUtil;
+import org.apache.skywalking.oap.log.analyzer.provider.LogAnalyzerModuleConfig;
+import org.apache.skywalking.oap.server.core.CoreModule;
+import org.apache.skywalking.oap.server.core.analysis.IDManager;
+import org.apache.skywalking.oap.server.core.analysis.NodeType;
+import org.apache.skywalking.oap.server.core.analysis.TimeBucket;
+import org.apache.skywalking.oap.server.core.config.NamingControl;
+import org.apache.skywalking.oap.server.core.source.EndpointMeta;
+import org.apache.skywalking.oap.server.core.source.ServiceInstanceUpdate;
+import org.apache.skywalking.oap.server.core.source.ServiceMeta;
+import org.apache.skywalking.oap.server.core.source.SourceReceiver;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+
+import static java.util.Objects.nonNull;
+
+/**
+ *
+ */
+@RequiredArgsConstructor
+public class TrafficAnalysisListener implements LogAnalysisListener {
+    private final SourceReceiver sourceReceiver;
+    private final NamingControl namingControl;
+
+    private ServiceMeta serviceMeta;
+    private ServiceInstanceUpdate instanceMeta;
+    private EndpointMeta endpointMeta;
+
+    @Override
+    public void build() {
+        if (nonNull(serviceMeta)) {
+            sourceReceiver.receive(serviceMeta);
+        }
+        if (nonNull(instanceMeta)) {
+            sourceReceiver.receive(instanceMeta);
+        }
+        if (nonNull(endpointMeta)) {
+            sourceReceiver.receive(endpointMeta);
+        }
+    }
+
+    @Override
+    public void parse(final LogData logData) {
+        long timeBucket;
+        if (logData.getTimestamp() == 0) {
+            timeBucket = TimeBucket.getRecordTimeBucket(System.currentTimeMillis());
+        } else {
+            timeBucket = TimeBucket.getRecordTimeBucket(logData.getTimestamp());
+        }
+        TimeBucket.getRecordTimeBucket(timeBucket);
+        // service
+        String serviceName = namingControl.formatServiceName(logData.getService());
```

Comment:
You seems assuming there will be service name in the log? I would say maybe not. We should add a check to ignore this kind of logs in the beginning.

Reply:
done